### PR TITLE
Login Enhancements and Exception Handling

### PR DESCRIPTION
**Pull request by caldempsey**
This PR:

1. Enables the login form to accept the "Enter" key for submission :)!!!
2. Provides visual feedback to the user by showing the login button as pressed when the Enter key is pressed.
3. Provides visual feedback when logins are successful.
4. Updates the login action to display a rate limit message if the user is being rate-limited.
5. Bubbles up the Exception from the login callback to provide better error handling and logging in the LoginUI to facilitate the above changes.

Please review and let me know of any feedback or changes needed. I am absolutely wasted and don't really know what the side effects were to these changes but I'm pretty sure its fine.

**Test Guidelines**

Please test the login flow: 

1. Click to login (valid)
2. Enter to login (valid)
3. Click to login (invalid)
4. Enter to login (invalid)

Invalid routes should trigger logging as expected. I capture all errors instead of just IOException et. al. since I'm unfamiliar with the codebase.

**Screenshots**

image

image

### CODE REVIEW

File: SwingUI/src/main/java/com/hawolt/ui/login/LoginUI.java [hunk 1]
```diff
@@ -50,15 +50,50 @@ private LoginUI(LeagueClientUI clientUI) {
...
+            if (e.getMessage().contains("RATE_LIMITED")) {
+                login.setText("You are being rate limited, try again later...");
+            } else {
+                login.setText("Login");
+            }
+            login.getModel().setPressed(false);
+            login.update(login.getGraphics());
+            Logger.error(e);
+            Logger.fatal(e);
+        }
```
Comment (caldempsey) [inline SwingUI/src/main/java/com/hawolt/ui/login/LoginUI.java:90]: Current flow is to recover the League client by calling Logger.fatal. Unsure why this works but seems the right place to recover the logger is in the logger UX. This lets us mutate the Login button to facilitate this unsolicited enhancement.

**Comment by slyk26** [comment]
change destination branch hawolt:main to hawolt:dev pls

**Comment by caldempsey** [comment]
@slyk26 Applied

**Comment by caldempsey** [comment]
Update: Added a success prompt when logins don't fail with an exception.

**Comment by hawolt** [comment]
I like the idea but I dont like the implementation when I have time I will see what changes I make ontop of yours on this

**Comment by caldempsey** [comment]
> I like the idea but I dont like the implementation when I have time I will see what changes I make ontop of yours on this

The problem is because the login JButton isn't decoupled from submit login there's not a lot of other ways to do this. I think the code could see some improvements. I avoided catching specific exceptions for lack of knowing which exceptions are relevant.

**Pull request merged**